### ISSUE TITLE
CI frontend

### ISSUE THREAD

**Issue by markus2330** [ElektraInitiative/PermaplanT#98]
CI features not related to deployment, to be executed as separate build job on our Docker agents:

- [ ] unit tests?
- [ ] reformatting?
- [ ] linter?
- [ ] import database in PR tests (scraper)

(? refers that we don't know how to do it yet, not that we don't want it)

**Comment by markus2330** [comment]
I moved some tasks to #63, here only the frontend tasks remain.

@badnames @aidnurs  can you help out here a bit which commands are needed to do all tests, building of docu, reformatting, etc.?

@lukashartl All the jobs should be executed in parallel (backend+frontend).

**Comment by aidnurs** [comment]
Hey guys,

> can you help out here a bit which commands are needed to do all tests, building of docu, reformatting, etc.?

I suppose `npm run build` executes the linter as well, right @buenaflor? I suggest to setup the linter so that it fails if the formatting doesn't correspond to our prettier configs in `prettier.config.js` instead of the pipeline formats every file automatically on the build itself, so the developer is aware of every change.

Additionally, we could add `npm run test` to `package.json` doing nothing for the time being, so we have the CD pipeline fixed and ready. 

Except for these 2 things, we should be good. If something additional regarding the topic comes to my mind, I'll let you know for sure.

**Comment by markus2330** [comment]
Have we already enabled [all the eslint-plugin-react-hooks rules](https://react.dev/learn/editor-setup#linting)?

Is the code currently formatted via https://prettier.io?

**Comment by buenaflor** [comment]
`npx lint-staged` should be the command if you take a look at `package.json` 

```
  "lint-staged": {
    "src/**/*.{js,jsx,ts,tsx,json,css,scss,md}": [
      "prettier --write"
    ],
    "src/**/*.{js,jsx,ts,tsx,json}": [
      "eslint --max-warnings=0"
    ]
  }

```

**Comment by Bushuo** [comment]
@buenaflor
I think we need two more scripts for that. If nothing is staged in the commit, `npx lint-staged` won't do anything
```zsh
# paulb in ~/Projects/uni/bakk/PermaplanT/frontend on git:authors o [12:37:19] 
$ npx lint-staged
→ No staged files found.
```

Usually I have scripts like this in my projects
```json
// package.json
...
"scripts": {
  "format": "prettier --write .",
   "lint": "eslint --cache --cache-location ./node_modules/.cache/eslint ."
}
...
```

I did not try this yet. I can earliest in the evening today.
